spec/lib/saml_idp/encryptor_spec.rb

```diff
@@ -32,5 +32,26 @@ module SamlIdp
 
       expect(encrypted_doc.remove_namespaces!.xpath('//KeyName')).to be_empty
     end
+
+    context 'gcm encryption algorithm' do
+      let(:encryption_opts) do
+        {
+          cert: Default::X509_CERTIFICATE,
+          block_encryption: 'aes256-gcm',
+          key_transport: 'rsa-oaep-mgf1p',
+        }
+      end
+
+      it 'encrypts XML' do
+        raw_xml = '<foo>bar</foo>'
+        encrypted_xml = subject.encrypt(raw_xml)
+        expect(encrypted_xml).not_to match 'bar'
+        encrypted_doc = Nokogiri::XML::Document.parse(encrypted_xml)
+        encrypted_data = Xmlenc::EncryptedData.new(encrypted_doc.at_xpath('//xenc:EncryptedData',
+                                                                          Xmlenc::NAMESPACES))
```

Comment:
is it worth doing an xml query for the specific xml namespace like `http://www.w3.org/2009/xmlenc11#aes256-gcm` from the other method what was edited?

Reply:
maybe a controller spec ([like this one](https://github.com/18F/saml_idp/blob/main/spec/lib/saml_idp/controller_spec.rb#L119))to ensure that that encryption/decryption of responses works with different encryption algos?

Reply:
Added specs. It doesn't seem like `tripledes-cbc` works at all due to a potential issue in upstream `xmlenc` or how `saml_idp` integrates with it, so I've removed it.